src/Workspaces/CSharp/Portable/CodeGeneration/ExpressionGenerator.cs

```diff
@@ -97,7 +94,7 @@ private static ExpressionSyntax GenerateNullLiteral()
 
             return type == null || type.IsReferenceType || type.IsPointerType() || type.IsNullable()
                 ? GenerateNullLiteral()
-                : SyntaxFactory.DefaultExpression(type.GenerateTypeSyntax());
+                : (ExpressionSyntax)CSharpSyntaxGenerator.Instance.DefaultExpression(type);
```

Comment:
:memo: This would have been better as a separate PR before the introduction of the new analyzer/fix.